### ISSUE TITLE
Allow to exclude weapons from dice progressions

### ISSUE THREAD

**Issue by Lichtblitz** [SoxMax/FG-Size-Changes#9]
Depending on the source of the effect that changes size, the weapons may or may not change size as well. Sometimes weapons are not even weapons, but rather work-arounds to certain spell damages. Currently, dice progressions seem to be applied to everything under "weapons".

It would be great to have a allow-/disallow flag for applying size changes per weapon. For example adding a damage type of `nosize` to a weapon could ignore dice progression on that weapon.

_Optional: In addition it would be great to have another optional descriptor on the SIZE effect, that would invert the behavior. For example the descriptor `noweapon` would only apply the dice progression to weapons with an explicit `size` damage type._

If you think this is a good feature for the extension, I might even give it a shot at implementing some of it.

**Comment by SoxMax** [comment]
Technically size changes dice progression applies to any damage role with the type of `damage` which sounds reductive but damage rolls can also have `spdamage`, which as far as I can tell is only used for spells. Also because of this it affects non-spell damage rolls from NPC sheets as well as PC sheets.
I'm not against adding a damage type to turn off size progression effects, but can you give an example of why you'd need to use a weapon to replicate a spell? As far as I was aware you could do everything with spell damage action that you can do with the weapon damage action, except for increasing the critical multiplier.

**Comment by Lichtblitz** [comment]
I would have to double check what exactly forced me to do the work-around, but I encountered it with the [Thorny Entanglement Spell](https://www.d20pfsrd.com/magic/all-spells/t/thorny-entanglement/). It has an attack roll and to avoid constantly scrolling up and down, I wanted to handle the damage through the weapon as well as the attack roll. However, the 2d6 got downgraded to 1d8 for my small druid with no option to disable the dice progression.
My DM and I prefer your extension over the alternatives and believe that with the suggested addition, it would cover all our use cases.

**Comment by SoxMax** [comment]
Feel free to open a pull request with the changes and I'll take a look.

**Comment by Lichtblitz** [comment]
I have implemented the main part of the issue. I'm not sure if the optional part is still needed, I'll figure this out as I use the extension more.
I have tested this with...
1. Weapons without `clauses`
2. Weapons with empty damage type
3. Weapons with some other damage type
4. Weapons with "nosize" damage type
5. Weapons with "nosize" damage type at the beginning, the end and middle of multiple types, with and without whitespace separating it from the commas

I only check the first clause, as this is the only one the dice progression is being applied to, anyways.
